enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,568 @@
+  - cybertron - networking
+  - zaneb - networking
+approvers:
+  - JoelSpeed
+  - patrickdillon
+  - cybertron
+api-approvers: 
+  - JoelSpeed
+creation-date: 2022-10-21
+last-updated: 2022-11-01
+tracking-link: 
+- https://issues.redhat.com/browse/OCPPLAN-9654
+see-also:
+  - /enhancements/network/baremetal-ipi-network-configuration.md
+  - /enhancements/installer/vsphere-ipi-zonal.md
+replaces:
+superseded-by:
+---
+
+# Static IP Addresses for vSphere IPI
+
+## Summary
+
+Static IP addresses are emerging as a common requirement in environments where
+the usage of DHCP violates corporate security guidelines.  Additionally, many 
+users which require static IPs also require the use of the IPI installer. 
+The proposal described in this enhacement discusses the implementation of
+of assiging static IPs at both day 0 and day 2.
+
+## Motivation
+
+Users of OpenShift would like the ability to provision vSphere IPI clusters with static IPs.
+
+- https://issues.redhat.com/browse/OCPPLAN-9654
+
+### User Stories
+
+As an OpenShift administrator, I want to provision nodes with static IP addresses so that I can comply with my organization's security requirements.
+
+As an OpenShift administrator, I want to provision static IP addresses with the IPI installer so that I can reduce the complexity of certifying tools required to provision OpenShift.
+
+As an OpenShift administrator, I want to scale nodes with static IPs so that I can meet capacity demands as well as respond to disaster recovery scenarios.
+
+### Goals
+
+- All nodes created during the installation are configured with static IPs
+      Rationale: Many environments, due to security policies, do not allow DHCP.
+
+- The IPI installation method is able to provide static IPs to the nodes
+  Rationale: Some users must qualify each tool used in their environment. 
+  Leveraging IPI greatly reduces the number of tools required to provision
+  a cluster.
+- 
+
+### Non-Goals
+
+- OpenShift will not be responsible for managing IP addresses.
+
+## Proposal
+
+### Static IPs Configured at Installation
+
+To faciliate the configuration of static IP address, nmstate definitions are created for each node in the install-config.yaml. This follows a similar pattern set forth in the [Baremetal IPI Network Configuration](https://github.com/openshift/enhancements/blob/master/enhancements/network/baremetal-ipi-network-configuration.md#user-facing-api) enhancement. 
+
+This enhancement expands on the [`hosts` slice](https://github.com/openshift/enhancements/blob/master/enhancements/network/baremetal-ipi-network-configuration.md#user-facing-api) by associating hosts with roles and, optionally, a failure domain.
+
+For the bootstrap and control plane nodes, static IPs are passed to the node via the `guestinfo.afterburn.initrd.network-kargs` extraconfig parameter.  [Afterburn](https://github.com/coreos/afterburn/blob/main/src/providers/vmware/amd64.rs) recognizes this parameter when the node initially boots. 
```

Comment:
It might be worth explicitly noting that the config passed in needs to be converted to kernel args. From an offline discussion it was not clear to everyone that there would be an nmstate to karg conversion happening under the covers.

I also say this because my other comment seems pretty out of left field if you don't realize that the end output of this config is going to be kargs regardless of what the API looks like. :-)

Reply:
That's a good point, I'll add a note on that.